server/src-lib/Hasura/GraphQL/Execute/LiveQuery/Poll.hs

```diff
@@ -64,7 +72,35 @@ import           Hasura.Session
 -- -------------------------------------------------------------------------------------------------
 -- Subscribers
 
-newtype Subscriber = Subscriber { _sOnChangeCallback :: OnChange }
+-- | Identifies a subscriber of a live query based on their (websocket id,
+-- operation id) - this is useful for collecting metrics and then correlating.
+-- The 'createdAt' to ensure that there is uniqueness even if the operationId
+-- is later repeated on the same websocket
```

Comment:
Actually, upon further thought, this seems sketchy to me. Suppose the operation id *is* repeated on the same websocket:

  1. Presumably, if we already have an active subscription for that operation id, we should reject the new one with an error, right?

  2. If that isn’t the case, and it’s okay to have multiple simultaneous subscriptions with the same operation id, then using `getCurrentTime` isn’t necessarily enough of a uniquifier, since it introduces a race condition if two subscriptions with the same operation id are created at the same time.

If uniqueness is required, it seems to me that it should either be enforced in the logic explicitly from the websocket and operation ids alone, or a `SubscriberId` should remain a UUID, and the value it’s *associated* with should hold the metadata for reporting purposes. Relying on a timestamp like this is bogus.